### PR TITLE
GEODE-8228: Fixes flaky WAN tests

### PR DESCRIPTION
**Pull request by pivotal-jbarrett**
### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### CODE REVIEW

File: geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/parallel/ParallelWANConflationDUnitTest.java [hunk 1]
```diff
@@ -120,26 +120,26 @@ public void testParallelPropagationBatchConflation() {
...
-        (ArrayList<Integer>) vm7.invoke(() -> WANTestBase.getSenderStats("ln", 100));
-    assertThat((v4List.get(10) + v5List.get(10) + v6List.get(10) + v7List.get(10)) == 100).as(
-        "Event in secondary queue should be 100").isTrue();
+    await().untilAsserted(() -> {
+      List<Integer> v4List = vm4.invoke(() -> WANTestBase.getSenderStats("ln", 100));
+      List<Integer> v5List = vm5.invoke(() -> WANTestBase.getSenderStats("ln", 100));
+      List<Integer> v6List = vm6.invoke(() -> WANTestBase.getSenderStats("ln", 100));
+      List<Integer> v7List = vm7.invoke(() -> WANTestBase.getSenderStats("ln", 100));
+      assertThat((v4List.get(10) + v5List.get(10) + v6List.get(10) + v7List.get(10)) == 100).as(
+          "Event in secondary queue should be 100").isTrue();
```
Comment (DonalEvans) [inline geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/parallel/ParallelWANConflationDUnitTest.java:129]: Could this be changed to `isEqualTo(100);` please?

File: geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/parallel/ParallelWANConflationDUnitTest.java [hunk 2]
```diff
@@ -217,38 +212,34 @@ public void doTestParallelPropagationConflation(int redundancy) {
...
+      List<Integer> vm4List = vm4.invoke(() -> WANTestBase.getSenderStats("ln", expectedNum));
+      List<Integer> vm5List = vm5.invoke(() -> WANTestBase.getSenderStats("ln", expectedNum));
+      List<Integer> vm6List = vm6.invoke(() -> WANTestBase.getSenderStats("ln", expectedNum));
+      List<Integer> vm7List = vm7.invoke(() -> WANTestBase.getSenderStats("ln", expectedNum));
+      assertThat(
+          (vm4List.get(10) + vm5List.get(10) + vm6List.get(10) + vm7List.get(10)) == expectedNum
+              * redundancy).as(
+                  "Event in secondary queue should be " + (expectedNum * redundancy) + ", but is "
+                      + (vm4List.get(10) + vm5List.get(10) + vm6List.get(10) + vm7List.get(10)))
+                  .isTrue();
```
Comment (DonalEvans) [inline geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/parallel/ParallelWANConflationDUnitTest.java:225]: Could this be changed to `isEqualTo(expectedNum * redundancy);` please?

File: geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/parallel/ParallelWANConflationDUnitTest.java [hunk 3]
```diff
@@ -120,26 +120,26 @@ public void testParallelPropagationBatchConflation() {
...
+    await().untilAsserted(() -> {
+      List<Integer> v4List = vm4.invoke(() -> WANTestBase.getSenderStats("ln", 0));
+      List<Integer> v5List = vm5.invoke(() -> WANTestBase.getSenderStats("ln", 0));
+      List<Integer> v6List = vm6.invoke(() -> WANTestBase.getSenderStats("ln", 0));
+      List<Integer> v7List = vm7.invoke(() -> WANTestBase.getSenderStats("ln", 0));
 
-    assertThat((v4List.get(8) + v5List.get(8) + v6List.get(8) + v7List.get(8)) > 0).as(
-        "No events conflated in batch").isTrue();
+      assertThat((v4List.get(8) + v5List.get(8) + v6List.get(8) + v7List.get(8)) > 0).as(
+          "No events conflated in batch").isTrue();
```
Comment (DonalEvans) [inline geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/parallel/ParallelWANConflationDUnitTest.java:141]: Could this be `isGreaterThan(0);` please?

File: geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/parallel/ParallelWANConflationDUnitTest.java [hunk 4]
```diff
@@ -217,38 +212,34 @@ public void doTestParallelPropagationConflation(int redundancy) {
...
+      List<Integer> vm5List = vm5.invoke(() -> WANTestBase.getSenderStats("ln", 0));
+      List<Integer> vm6List = vm6.invoke(() -> WANTestBase.getSenderStats("ln", 0));
+      List<Integer> vm7List = vm7.invoke(() -> WANTestBase.getSenderStats("ln", 0));
+      assertThat(
+          (vm4List.get(11) + vm5List.get(11) + vm6List.get(11) + vm7List.get(11)) == expectedNum
+              * redundancy).as(
+                  "Event processed by queue removal message should be " + (expectedNum * redundancy)
+                      + ", but is "
+                      + (vm4List.get(11) + vm5List.get(11) + vm6List.get(11) + vm7List.get(11)))
+                  .isTrue();
```
Comment (DonalEvans) [inline geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/parallel/ParallelWANConflationDUnitTest.java:241]: Could this be changed to `isEqualTo(expectedNum * redundancy);` please?

File: geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/parallel/ParallelWANStatsDUnitTest.java [hunk 5]
```diff
@@ -297,27 +301,25 @@ public void testPartitionedRegionParallelPropagation_AfterDispatch_NoRedundancy(
...
+      // queue size:
+      assertThat(v4List.get(0) + v5List.get(0) + v6List.get(0) + v7List.get(0)).isEqualTo(0);
+      // eventsReceived:
+      assertThat(v4List.get(1) + v5List.get(1) + v6List.get(1) + v7List.get(1)).isEqualTo(NUM_PUTS);
+      // events queued:
+      assertThat(v4List.get(2) + v5List.get(2) + v6List.get(2) + v7List.get(2)).isEqualTo(NUM_PUTS);
+      // events distributed:
+      assertThat(v4List.get(3) + v5List.get(3) + v6List.get(3) + v7List.get(3)).isEqualTo(NUM_PUTS);
+      // batches distributed:
+      assertThat(v4List.get(4) + v5List.get(4) + v6List.get(4) + v7List.get(4) >= 10).isTrue();
```
Comment (DonalEvans) [inline geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/parallel/ParallelWANStatsDUnitTest.java:319]: Could this be `isGreaterThanOrEqualTo(10);` please?

File: geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/parallel/ParallelWANStatsDUnitTest.java [hunk 6]
```diff
@@ -718,21 +716,22 @@ public void testPRParallelPropagationWithBatchRedistWithoutGroupTransactionEvent
...
+      // events received:
+      assertThat((int) v4List.get(1)).isEqualTo(entries);
+      // events queued:
+      assertThat((int) v4List.get(2)).isEqualTo(entries);
+      // events distributed:
+      assertThat((int) v4List.get(3)).isEqualTo(entries);
+      // batches distributed:
+      assertThat((int) v4List.get(4)).isEqualTo(3);
+      // batches redistributed:
+      assertThat((v4List.get(5)) > 0).as("Batch was not redistributed").isTrue();
```
Comment (DonalEvans) [inline geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/parallel/ParallelWANStatsDUnitTest.java:733]: Could this be `isGreaterThan(0);` please?

File: geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/parallel/ParallelWANStatsDUnitTest.java [hunk 7]
```diff
@@ -793,23 +792,24 @@ public void testPRParallelPropagationWithBatchRedistWithGroupTransactionEventsSe
...
+      // events received:
+      assertThat((int) v4List.get(1)).isEqualTo(entries);
+      // events queued:
+      assertThat((int) v4List.get(2)).isEqualTo(entries);
+      // events distributed:
+      assertThat((int) v4List.get(3)).isEqualTo(entries);
+      // batches distributed:
+      assertThat((int) v4List.get(4)).isEqualTo(2);
+      // batches redistributed:
+      assertThat((v4List.get(5)) > 0).as("Batch was not redistributed").isTrue();
```
Comment (DonalEvans) [inline geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/parallel/ParallelWANStatsDUnitTest.java:809]: Could this be `isGreaterThan(0);` please?

File: geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/parallel/ParallelWANStatsDUnitTest.java [hunk 8]
```diff
@@ -832,27 +832,25 @@ public void testPartitionedRegionParallelPropagation_AfterDispatch_Redundancy_3(
...
+      // queue size:
+      assertThat(v4List.get(0) + v5List.get(0) + v6List.get(0) + v7List.get(0)).isEqualTo(0);
+      // events received:
+      assertThat(v4List.get(1) + v5List.get(1) + v6List.get(1) + v7List.get(1)).isEqualTo(400);
+      // events queued:
+      assertThat(v4List.get(2) + v5List.get(2) + v6List.get(2) + v7List.get(2)).isEqualTo(400);
+      // events distributed
+      assertThat(v4List.get(3) + v5List.get(3) + v6List.get(3) + v7List.get(3)).isEqualTo(NUM_PUTS);
+      // batches distributed:
+      assertThat(v4List.get(4) + v5List.get(4) + v6List.get(4) + v7List.get(4) >= 10).isTrue();
```
Comment (DonalEvans) [inline geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/parallel/ParallelWANStatsDUnitTest.java:850]: Could this be `isGreaterThanOrEqualTo(10);` please?

**Review by DonalEvans — APPROVED** [review]
Nice clean-up. A few nice-to-have suggestions, but nothing that's a deal-breaker given how much of an improvement this is over what was there previously.

File: geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderAlertThresholdDUnitTest.java [hunk 9]
```diff
@@ -86,17 +86,16 @@ public void testParallelSenderQueueEventsAlertThreshold() {
...
+      List<Integer> v4List = vm4.invoke(() -> WANTestBase.getSenderStats("ln", -1));
+      List<Integer> v5List = vm5.invoke(() -> WANTestBase.getSenderStats("ln", -1));
+      List<Integer> v6List = vm6.invoke(() -> WANTestBase.getSenderStats("ln", -1));
+      List<Integer> v7List = vm7.invoke(() -> WANTestBase.getSenderStats("ln", -1));
 
-    assertThat((v4List.get(12) + v5List.get(12) + v6List.get(12) + v7List.get(12)) > 0).as(
-        "GatewaySenders Stats should contain number of EventsExceedingAlertThreshold > 0").isTrue();
+      assertThat((v4List.get(12) + v5List.get(12) + v6List.get(12) + v7List.get(12)) > 0).as(
+          "GatewaySenders Stats should contain number of EventsExceedingAlertThreshold > 0")
+          .isTrue();
```
[the remark this answers is not in the document]
Comment (pivotal-jbarrett) [reply geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/parallel/ParallelGatewaySenderAlertThresholdDUnitTest.java:97]: I fixed a few of these, not sure how I missed some of these others. Thanks for pointing them out.

I get what you are saying in the second option but we can't assume that the stat in any one of those collections is actually greater than 0 but that the sum of them all is.

File: geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/Simple2CacheServerDistributedTest.java [hunk 10]
```diff
@@ -63,11 +63,11 @@ public void doMultipleCacheServer(boolean durable) throws Exception {
...
 
     if (durable) {
-      vm2.invoke(Simple2CacheServerDUnitTest::setCacheClientProxyTestHook);
+      vm2.invoke(Simple2CacheServerDistributedTest::setCacheClientProxyTestHook);
     } else {
-      vm3.invoke(Simple2CacheServerDUnitTest::setClientServerObserver);
+      vm3.invoke(Simple2CacheServerDistributedTest::setClientServerObserver);
     }
-    vm3.invoke(() -> CacheClientNotifierDUnitTest.createClientWithLocator(lnPort, "localhost",
+    vm3.invoke(() -> CacheClientNotifierDistributedTest.createClientWithLocator(lnPort, "localhost",
```
Comment (kirklund) [inline geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/Simple2CacheServerDistributedTest.java:70]: Might want to extract `createClientWithLocator` to a new helper class to untangle these two distributed tests.

File: geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/Simple2CacheServerDistributedTest.java [hunk 11]
```diff
@@ -76,24 +76,25 @@ public void doMultipleCacheServer(boolean durable) throws Exception {
...
       return checkProxyIsPrimary(vm0) || checkProxyIsPrimary(vm2);
     });
 
     // close the current primary cache server, then re-test
-    int serverPortAtVM1 = vm2.invoke(Simple2CacheServerDUnitTest::findCacheServerForPrimaryProxy);
+    int serverPortAtVM1 =
+        vm2.invoke(Simple2CacheServerDistributedTest::findCacheServerForPrimaryProxy);
     if (serverPortAtVM1 != 0) {
-      vm2.invoke(() -> CacheClientNotifierDUnitTest.closeACacheServer(serverPortAtVM1));
+      vm2.invoke(() -> CacheClientNotifierDistributedTest.closeACacheServer(serverPortAtVM1));
```
Comment (kirklund) [inline geode-wan/src/distributedTest/java/org/apache/geode/internal/cache/wan/Simple2CacheServerDistributedTest.java:91]: Here's another helper method to potentially extract.

**Review by kirklund — APPROVED** [review]

**Review by nabarunnag — APPROVED** [review]

**Pull request merged**